reference/src/layout/arrays-and-slices.md

```diff
@@ -0,0 +1,86 @@
+# Layout of Rust array types and slices
+
+## Layout of Rust array types 
+
+Array types, `[T; N]`, store `N` values of type `T` with a constant _stride_.
+Here, _stride_ is the distance between each pair of consecutive values within
+the array.
+
+The _offset_ of the first array element is `0`, that is, a pointer to the array
+and a pointer to its first element both point to the same memory address.
+
+The _alignment_ of array types is greater or equal to the alignment of its
+element type. If the element type is `repr(C)` the layout of the array is
+guaranteed to be the same as the layout of a C array with the same element type.
+
+> **Note**: the type of array arguments in C function signatures, e.g., `void
+> foo(T x[N])`, decays to a pointer. That is, these functions do not take arrays
+> as an arguments, they take a pointer to the first element of the array
+> instead. Array types are therefore _improper C types_ (not C FFI safe) in Rust
+> foreign function declarations, e.g., `extern { fn foo(x: [T; N]) -> [U; M];
+> }`. Pointers to arrays are fine: `extern { fn foo(x: *const [T; N]) -> *const
+> [U; M]; }`.
+
+The _stride_ of the array is the distance between a pair of consecutive values
```

Comment:
This repeats the first sentence, which reads a bit strange to me since they're not that far apart. I'm not sure if omitting this and just stating how the stride is computed here is any better?

Reply:
I've changed it a bit. Is it better now?